api/src/main/java/com/epam/pipeline/manager/cloud/azure/AzureInstanceService.java

```diff
@@ -240,6 +243,10 @@ private String buildNodeUpCommand(final AzureRegion region, final Long runId, fi
                 .kubeToken(kubeToken)
                 .region(region.getRegionCode());
 
+        final Boolean clusterSpotStrategy = preferenceManager.getPreference(SystemPreferences.CLUSTER_SPOT);
```

Comment:
As far as I remember, we shall use preference value only if `instance.getSpot() == null`

Reply:
correct